fabtests/functional/rdm_stress.c

```diff
@@ -593,58 +592,280 @@ static int run_child(void)
+	int obj_size; /* # child tokens in object token */
+	const char *key_str;
+	jsmntok_t *cur_token;
+	size_t token_len;
+
+	assert(token[*idx].type == JSMN_OBJECT);
+
+	init_rpc_ctrl(ctrl);
+	obj_idx = *idx;
+	obj_size = token[*idx].size;
+
+	/* i is indexing # of key:value pairs in JSMN_OBJECT */
+	for (int i = 0; i < obj_size && *idx < num_tokens; i++) {
+		(*idx)++; /* advance to next token, expecting key token */
+		cur_token = &token[*idx];
+		token_len = cur_token->end - cur_token->start;
+		/* key token must be JSMN_STRING and size == 1 */
+		if (cur_token->type != JSMN_STRING || cur_token->size != 1)
+			goto err_out;
+
+		key_str = &json_payload[cur_token->start];
+		if (!strncmp(key_str, "op", token_len)) {
+			(*idx)++; /* advance to value token */
+			cur_token = &token[*idx];
+			if (!get_op_enum(json_payload, cur_token, &ctrl->op))
+				goto err_out;
+		} else if (!strncmp(key_str, "size", token_len) ||
+			   !strncmp(key_str, "ms", token_len)) {
+			(*idx)++; /* advance to value token */
+			cur_token = &token[*idx];
+			if (!get_uint64_val(json_payload, cur_token, &ctrl->size))
+				goto err_out;
+		} else if (!strncmp(key_str, "offset", token_len)) {
+			(*idx)++; /* advance to value token */
+			cur_token = &token[*idx];
+			if (!get_uint64_val(json_payload, cur_token, &ctrl->offset))
+				goto err_out;
+		} else {
+			goto err_out;
+		}
+	}
+
+	/* op is rquired for rpc_ctrl to be valid */
+	if (ctrl->op == 0xffffffff)
+		goto err_out;
+
+	return true;
+
+err_out:
+	printf("Invalid JSON entry: %.*s\n",
+		token[obj_idx].end - token[obj_idx].start,
+		&json_payload[token[obj_idx].start]);
+	init_rpc_ctrl(ctrl); /* in case this is the last token */
+
+	return false;
+}
+
+/* read and parse control file */
+static int init_ctrls(const char *ctrlfile)
+{
+	FILE *ctrl_f;
+	struct stat statbuf;
+	char *json_payload;	/* control file loaded in string */
+	int num_tokens;		/* # of JSON tokens in the control file */
+	int num_obj;		/* # of JSON objects = possible rpc_ctrl entries */
+	jsmn_parser parser;
+	jsmntok_t *tokens;
+	int start, i;
+	int ret = 0;
+
+	ctrl_f = fopen(ctrlfile, "r");
+	if (!ctrl_f)
+		return -errno;
+
+	if (stat(ctrlfile, &statbuf))
+		return -errno;
+
+	json_payload = malloc(statbuf.st_size + 1);
+	if (!json_payload) {
+		ret = -FI_ENOMEM;
+		goto no_mem_out;
+	}
+
+	if (fread(json_payload, statbuf.st_size, 1, ctrl_f) != 1) {
+		ret = -FI_EINVAL;
+		goto read_err_out;
+	}
+	json_payload[statbuf.st_size] = 0;
+
+	/* get # of tokens, allcoate memory and parse JSON */
+	jsmn_init(&parser);
+	num_tokens = jsmn_parse(&parser, json_payload, statbuf.st_size, NULL, 0);
+	if (num_tokens < 0) {
+		ret = -FI_EINVAL;
+		goto read_err_out;
+	}
+
+	tokens = malloc(sizeof(jsmntok_t) * num_tokens);
+	if (!tokens) {
+		ret = -FI_ENOMEM;
+		goto read_err_out;
+	}
+
+	jsmn_init(&parser);
+	if (jsmn_parse(&parser, json_payload, statbuf.st_size, tokens, num_tokens) != num_tokens) {
```

Comment:
limit line lengths to 80ish characters.  There are a few places where line lengths are long.

Reply:
i will make variable names shorter